tests/ShadowDOM/html/activeElement.html

```diff
@@ -0,0 +1,344 @@
+<script src="../../../../web-component-tester/browser.js"></script>
+
+<script>
+
+suite('activeElement', function() {
+  var wrap = ShadowDOMPolyfill.wrap;
+
+
+  function inflateChildren(node) {
+    var nodesForInflation = node.querySelectorAll('[shadow-template-id]');
+    Array.prototype.slice.apply(nodesForInflation).forEach(inflate);
+  }
+
+  function inflate(node) {
+    node.tabIndex = -1;
+
+    var shadowTemplateId = node.getAttribute('shadow-template-id');
+    if (shadowTemplateId) {
+      var template = document.getElementById(shadowTemplateId);
+      var templateClone = document.importNode(template.content, true);
+      inflateChildren(templateClone);
+      node.createShadowRoot().appendChild(templateClone);
+    }
+
+    inflateChildren(node);
+  }
+
+
+  var doc;
+  var r;
+    // light
+      var r_l;
+    // shadow
+      var r_0;
+        // light
+          var r_0_l;
+        // shadow
+          var r_0_0;
+            // light
+              var r_0_0_l;
+          var r_0_1;
+            // light
+              var r_0_1_l;
+      var r_1;
+        // light
+          var r_1_l;
+        // shadow
+          var r_1_0;
+            // light
+              var r_1_0_l;
+          var r_1_1;
+            // light
+              var r_1_1_l;
+
+  setup(function() {
+    doc = wrap(document);
+
+    r = document.createElement('div');
+    r.setAttribute('shadow-template-id', 'x-shadow-host-root');
+
+    r_l = document.createElement('div');
+    r_l.setAttribute('shadow-template-id', 'x-shadow-host-root-light');
+    r.appendChild(r_l);
+
+    inflate(r);
+
+    r_0 = r.shadowRoot.querySelector('[shadow-template-id=x-shadow-host-root-0]');
+      r_0_l = r_0.querySelector('[shadow-template-id=x-shadow-host-root-0-light]');
+      r_0_0 = r_0.shadowRoot.querySelector('[shadow-template-id=x-shadow-host-root-0-0]');
+        r_0_0_l = r_0_0.querySelector('[shadow-template-id=x-shadow-host-root-0-0-light]');
+      r_0_1 = r_0.shadowRoot.querySelector('[shadow-template-id=x-shadow-host-root-0-1]');
+        r_0_1_l = r_0_1.querySelector('[shadow-template-id=x-shadow-host-root-0-1-light]');
+    r_1 = r.shadowRoot.querySelector('[shadow-template-id=x-shadow-host-root-1]');
+      r_1_l = r_1.querySelector('[shadow-template-id=x-shadow-host-root-1-light]');
+      r_1_0 = r_1.shadowRoot.querySelector('[shadow-template-id=x-shadow-host-root-1-0]');
+        r_1_0_l = r_1_0.querySelector('[shadow-template-id=x-shadow-host-root-1-0-light]');
+      r_1_1 = r_1.shadowRoot.querySelector('[shadow-template-id=x-shadow-host-root-1-1]');
+        r_1_1_l = r_1_1.querySelector('[shadow-template-id=x-shadow-host-root-1-1-light]');
+
+    document.body.appendChild(r);
+
+    // Nodes need to be in the document to be focusable.
+    ShadowDOMPolyfill.renderAllPending();
+  });
+
+  teardown(function() {
+    document.body.removeChild(r);
+
+    assert.equal(r_0_0.shadowRoot.activeElement, null);
```

Comment:
These assertions don't work because the elements aren't attached to the document after the `removeChild` call above - I'll move it to the end. Also, are assertions within the post-test hook ok?